modules/finance/src/main/java/com/opengamma/strata/finance/rate/swaption/Swaption.java

```diff
@@ -0,0 +1,777 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.finance.rate.swaption;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableValidator;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.LongShort;
+import com.opengamma.strata.basics.date.BusinessDayAdjustment;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.finance.rate.swap.Swap;
+
+/**
+ * An option on an underlying swap.
+ * <p>
+ * A swaption is a financial instrument that provides an option based on the future value of a swap.
+ * The option is European, exercised only on the exercise date.
+ */
+@BeanDefinition
+public final class Swaption
+    implements SwaptionProduct, ImmutableBean, Serializable {
+
+  /**
+   * Whether the option is long or short.
+   * <p>
+   * Long indicates that the owner wants the option to be in the money at expiry.
+   * Short indicates that the owner wants the option to be out of the money at expiry.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final LongShort longShort;
+  /**
+   * Settlement method.  
+   * <p>
+   * The settlement of the option is specified by {@link SwaptionSettlementMethod}.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final SwaptionSettlementMethod settlementMethod;
+  /**
+   * The expiry date of the option.  
+   * <p>
+   * The option is European, and can only be exercised on the expiration date. 
+   * <p>
+   * This date is typically set to be a valid business day.
+   * Optionally, the {@code businessDayAdjustment} property may be set to provide a rule for adjustment.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final LocalDate expiryDate;
```

Comment:
Use `AdjustableDate` instead of `LocalDate` and remove `businessDayAdjustment` field.